pkg/arvo/tests/sys/hoon/list.hoon

```diff
@@ -0,0 +1,274 @@
+    %+  expect-eq
+      !>  ~
+      !>  (fand `(list @)`~[6] `(list @)`~[1 1 1 2 3])
+  ==
+::
+++  test-list-find
+  =/  l=(list @)  ~[1 1 1 2 3]
+  ;:  weld
+    ::  Checks checks looking for a single item
+    ::
+    %+  expect-eq
+      !>  (some 0)
+      !>  (find `(list @)`~[1] l)
+    ::  checks looking for a sub list
+    ::
+    %+  expect-eq
+      !>  (some 2)
+      !>  (find `(list @)`~[1 2 3] l)
+    ::  Checks when the value is not present
+    ::
+    %+  expect-eq
+      !>  ~
+      !>  (find `(list @)`~[6] l)
+  ==
+::
+++  test-list-flop
+  ;:  weld
+    ::  A list reversed 2x is the same list
+    ::
+    %+  expect-eq
+      !>  `(list @)`~[1 1 1 2 3]
+      !>  (flop (flop `(list @)`~[1 1 1 2 3]))
+    :: An empty list reversed is still empty
+    ::
+    %+  expect-eq
+      !>  ~
+      !>  (flop `(list @)`~)
+  ==
+::
+++  test-list-gulf
+  ::  check for an increasing list
+  ::
+  %+  expect-eq
+    !>  ~[0 1 2 3]
+    !>  (gulf 0 3)
+::
+++  test-list-join
+  %+  expect-eq
+    !>  ~[2 1 3]
+    !>  (join 1 `(list @)`~[2 3])
+::
+++  test-list-lent
+  ;:  weld
+    %+  expect-eq
+      !>  2
+      !>  (lent `(list @)`~[2 3])
+    %+  expect-eq
+      !>  0
+      !>  (lent *(list @))
+  ==
+::
+++  test-list-levy
+  =/  l=(list @)  ~[2 3]
+  ;:  weld
+    :: Tail fails condition
+    ::
+    %+  expect-eq
+      !>  %.n
+      !>  (levy l |=(x=@ (gth 3 x)))
+    :: Head fails condition
+    ::
+    %+  expect-eq
+      !>  %.n
+      !>  (levy l |=([x=@] (lth 2 x)))
+    :: Both pass condition
+    ::
+    %+  expect-eq
+      !>  %.y
+      !>  (levy l |=([x=@] (lth x 10)))
+  ==
+::
+++  test-list-lien
+  =/  l=(list @)  ~[2 3]
+  ;:  weld
+    :: only Tail fails condition
+    ::
+    %+  expect-eq
+      !>  %.y
+      !>  (lien l |=(x=@ (gth 3 x)))
+    :: only head fails condition
+    ::
+    %+  expect-eq
+      !>  %.y
+      !>  (lien l |=([x=@] (lth 2 x)))
+    :: Both pass condition
+    ::
+    %+  expect-eq
+      !>  %.y
+      !>  (lien l |=([x=@] (lth x 10)))
+      :: Neither
+      ::
+    %+  expect-eq
+      !>  %.y
+      !>  (lien l |=([x=@] (gth 10 x)))
+  ==
+::
+++  test-list-murn
+  %+  expect-eq
+    !>  ~[6 10]
+    !>  %+  murn  `(list @)`~[2 3 4 5] 
+          |=  [x=@] 
+          ^-  (unit)
+          ?:  =((mod x 2) 0)
+            [~]
+          (some (mul x 2))
+::
+++  test-list-oust
+  ;:  weld
+    %+  expect-eq
+      !>  ~[2 5]
+      !>  (oust [1 2] `(list @)`~[2 3 4 5])
+    %+  expect-eq
+      !>  ~[2 3 4 5]
+      !>  (oust [1 0] `(list @)`~[2 3 4 5])
+  ==
+::
+++  test-list-reap
+  ;:  weld
+    %+  expect-eq
+      !>  ~[1 1 1]
+      !>  (reap 3 1)
+  ==
+::
+++  test-list-reel
+  :: Use non-associtive operation
+  ::
+  %+  expect-eq
+    !>  93
+    !>  (reel `(list @)`~[100 10 5 2] sub)
+::
+++  test-list-roll
+  :: Use non-associative operation
+  %+  expect-eq
+    !>  6
+    !>  %+  roll  `(list @)`~[1 5 10] 
+          |=  [acc=@ x=@] 
+          ?~  acc  x
```

Comment:
The null check is superfluous, this should just be `(roll `(list @)`~[1 5 10] sub)`.